phoenix-core/src/main/java/org/apache/phoenix/compile/ScanRanges.java

```diff
@@ -385,9 +387,25 @@ public Scan intersectScan(Scan scan, final byte[] originalStartKey, final byte[]
         if (scanStopKey.length > 0 && Bytes.compareTo(scanStartKey, scanStopKey) >= 0) { 
             return null; 
         }
-        newScan.setAttribute(SCAN_ACTUAL_START_ROW, scanStartKey);
-        newScan.setStartRow(scanStartKey);
-        newScan.setStopRow(scanStopKey);
+        if(ScanUtil.isLocalIndex(scan)) {
```

Comment:
@JamesRTaylor  we cannot use keyoffset > 0 check always for considering local indexes scan because in special case like if a table has only one region then region start key length and end key length is zero so keyoffset also becoming zero. In that case we need to check local index scan or not to set the attributes properly. Wdyt?